### ISSUE TITLE
Use of Moshi over Gson

### ISSUE THREAD

**Issue by PatilShreyas** [nic-delhi/AarogyaSetu_Android#23]
Moshi is a modern, high performance and recommend serialisation library for kotlin and java by Square developers. Gson has some limitations and existing bugs which Moshi overcomes. If it's used in this codebase we can also reduce some amount of code.

**Comment by crowded-geek** [comment]
@PatilShreyas I don't think there's any issue with using Gson. If so please mention exactly where and how to reproduce the issue.

**Comment by PatilShreyas** [comment]
Yeah. No issues with current implementation but it's for enhancement and best practices.

**Comment by Apsaliya** [comment]
Gson with Kotlin is a big no-no. Largely due to fact that it does not understand kotlin language features such as nullability, default values. Moshi has very good kotlin support with little setup but it's straightforward and should be given preference any day.

**Comment by bpanigrahi** [comment]
I don't see any issue with using Gson. Even GSON  passes green in SonarQube scan as well.
